### PR TITLE
Don't call auto_increment for columns where it's not set

### LINKED ISSUE #1357: Hitting 'negative last_insert_rowid' panic with Sqlite
## Description

I have a table with a primary ID that is a hash, stored as a signed integer (since unsigned 64 bit is not supported). Auto-increment is turned off for that table.
When storing a new element, if the hash as signed is negative, then sea-orm panics on that line: https://github.com/SeaQL/sea-orm/blob/master/src/executor/execute.rs#L43

## Steps to Reproduce

- Create an SQlite table with an i64 primary key: https://github.com/nitnelave/lldap/blob/c64d32e2c0cc698c61ca40e8364ab0cd7934cd46/server/src/domain/model/jwt_refresh_storage.rs#L12
- Insert a negative value.

### Expected Behavior

Since auto-increment is turned off for that table, `last_insert_id` should probably not be called to avoid a panic.

### Actual Behavior

Panic.

### Reproduces How Often

Always.

## Versions

Using sea-orm 0.10.3 (no other version override). Running on alpine, using the bundled Sqlite.

## Additional Information

The query, for reference: https://github.com/nitnelave/lldap/blob/c64d32e2c0cc698c61ca40e8364ab0cd7934cd46/server/src/infra/sql_backend_handler.rs#L49

### PR DESCRIPTION
**Pull request by nitnelave**
Fixes #1357

### CODE REVIEW

**Comment by nitnelave** [comment]
I didn't find a simple way to add a test, since the bakery schema doesn't have any model with an integer, non-auto-increment primary key and it doesn't seem easy to add a custom schema for a test. If you have ideas or hints on how to add the tests, I can do that.

**Comment by tyt2y3** [comment]
Umm no clear reason why the tests failed

**Comment by tyt2y3** [comment]
I see why the test fail here.

File: src/executor/insert.rs [hunk 1]
```diff
@@ -137,28 +137,30 @@ where
...
-                .map(|col| col.to_string())
-                .collect::<Vec<_>>();
-            let rows = db.query_all(statement).await?;
-            let res = rows.last().ok_or_else(|| {
-                DbErr::RecordNotInserted("None of the records are being inserted".to_owned())
-            })?;
-            res.try_get_many("", cols.as_ref()).ok()
-        }
-        false => {
-            let last_insert_id = db.execute(statement).await?.last_insert_id();
```
Comment (tyt2y3) [inline src/executor/insert.rs:152]: We should still execute the statement here, but just not trying to call `last_insert_id` from it unless it's really needed.

**Comment by tyt2y3** [comment]
@billy1624 can you revise the implementation?

File: src/executor/insert.rs [hunk 2]
```diff
@@ -137,28 +137,30 @@ where
...
-            ValueTypeOf::<A>::try_from_u64(last_insert_id).ok()
```
Comment (tyt2y3) [inline src/executor/insert.rs:153]: We should also defer this `try_from_u64`

File: src/executor/insert.rs [hunk 3]
```diff
@@ -137,28 +137,30 @@ where
...
-        }
-    };
     let last_insert_id = match primary_key {
         Some(value_tuple) => FromValueTuple::from_value_tuple(value_tuple),
```
Comment (tyt2y3) [inline src/executor/insert.rs:141]: To make it consistent with `support_returning = true` path

**Comment by tyt2y3** [comment]
Close in favour of #1403

**Pull request closed without merging**